hahomematic/platforms/custom/cover.py

```diff
@@ -316,24 +364,32 @@ async def close(self, collector: CallParameterCollector | None = None) -> None:
             collector=collector,
         )
 
+    async def stop(self, collector: CallParameterCollector | None = None) -> None:
+        """Stop the device if in motion."""
+        async with self._command_processing_lock:
+            await self._stop(collector=collector)
+
     @bind_collector(use_command_queue=False)
+    async def _stop(self, collector: CallParameterCollector | None = None) -> None:
+        """Stop the device if in motion. Do only call with _command_processing_lock held."""
+        self.central.command_queue_handler.empty_queue(device_address=self.device.device_address)
+        await super().stop()
+
     async def open_tilt(self, collector: CallParameterCollector | None = None) -> None:
         """Open the tilt."""
         if not self.is_state_change(tilt_open=True):
             return
         await self._set_level(tilt_level=self._open_tilt_level, collector=collector)
 
-    @bind_collector(use_command_queue=False)
     async def close_tilt(self, collector: CallParameterCollector | None = None) -> None:
         """Close the tilt."""
         if not self.is_state_change(tilt_close=True):
             return
         await self._set_level(tilt_level=self._closed_level, collector=collector)
 
-    @bind_collector()
     async def stop_tilt(self, collector: CallParameterCollector | None = None) -> None:
-        """Stop the device if in motion."""
-        await self._e_stop.send_value(value=True, collector=collector)
+        """Stop the device if in motion. Use only when command_processing_lock is held."""
+        await self.stop(collector=collector)
 
     def is_state_change(self, **kwargs: Any) -> bool:
```

Comment:
This could be made aware of target positions

Reply:
Ich habe das mal versucht - so richtig glücklich war ich mit dem Ergebnis nicht. Das Problem ist, dass sowohl die einzelnen Methoden (wie `open`, `close`, `open_tilt`, `vent`, ...) als auch `is_state_change` unabhängig voneinander bestimmen, wie Target Level (und Target Tilt Level) abhängig von der auszuführenden Aktion zu setzen sind. Wenn hier noch eine Überprüfung von `target_level` und `target_tilt_level` dazukommt, wird es schnell sehr unübersichtlich.

Aus meiner Sicht wäre es sinnvoll, das Ableiten der Ziellevel und die Überprüfung, ob für gegebene Ziellevel ein neues Kommando gesendet werden muss, klar zu separieren. Dann kann man das Ableiten der Ziellevel auch an einer Stelle tun. Das wäre aber auch in nem eigenen PR ganz gut aufgehoben - was meinst du, @SukramJ?